packages/nitro-protocol/src/contract/transaction-creators/nitro-adjudicator.ts

```diff
@@ -13,7 +13,8 @@ export function createPushOutcomeTransaction(
   finalizesAt: number,
   state: State,
   outcome: Outcome,
-  channelWasConcluded = false
+  channelWasConcluded = false,
+  transferAll = false
```

Comment:
It seems like this is passing on some additional logic into the `nitro-protocol` package that we may not want to put in there. Should it be the `nitro-protocol` packages job to decide what transaction to construct or the `ChainService`'s?

Some points worth considering:
- The method is named `createPushOutcomeTransaction`, but sometimes returns a `pushOutcomeAndTransferAll` transaction.
- The `ChainService` has all of the information and context it needs to decide which transaction to construct; it could decide to call `createPushOutcomeTransaction`, `createPushOutcomeAndTransferAllTransaction`, or `createConcludeAndPushOutcomeAndTransferAllTransaction`, etc based on configured options / logic inside the `ChainService`, and the logic inside `nitro-protocol` could be simplified.

Ultimately seems like a question about separation of concerns that could go either way. Seems to me like it's "cleaner" to have the `nitro-protocol` expose TS helpers to simplify making known external API calls on the `NitroAdjudicator` contract as opposed to abstracting those methods, but that's just my two cents

Reply:
I think it's a good call to clean up the Nitro protocol API.

Reply:
da452420e